### DISCUSSION TITLE
Where is the config for textcat loss function and how can I use a custom log?

### DISCUSSION THREAD

**Discussion by samehraban** [explosion/spaCy#6619] [Help: Coding & Implementations]
## Your Environment

* **spaCy version:** 2.3.0.dev0
* **Platform:** Linux-5.4.0-58-generic-x86_64-with-glibc2.29
* **Python version:** 3.8.5

I've spent a lot of time digging spacy and thinc repositories but failed to find the config section which defined loss function for a model. There are some loss definitions in thinc but they weren't used anywhere in spacy (or I couldn't find them).

**Comment by svlandeg** [comment]
Hi! I converted your issue to a thread on our new discussion board, as we're trying to keep the issue tracker more focused on bug reports. This new discussion board also has a more "StackOverflow feel" to it which I kind of like, and it's more suited for the community to come together to discuss things and help eachother etc :-)

Anyway, the loss functions can't easily be redefined unless by subclassing an existing component. For spaCy v2, they are defined in [`pipes.pyx`](https://github.com/explosion/spaCy/blob/master/spacy/pipeline/pipes.pyx), e.g. the loss for the textcat is defined here: https://github.com/explosion/spaCy/blob/master/spacy/pipeline/pipes.pyx#L1041

**Reply by samehraban** [reply]
What about version 3? Did it make things any more simple?
